### PR TITLE
test automation

### PR DESCRIPTION
**Pull request by ejannett**
This commit will bring new testing features:
- small framework to handle tests configuration 
- small framework to record and play back any command line 
- new setuptool command to automatically create an OCI instance and run  oci_utils test on it.

instructions :  see doc/dev-notes

Note : 
This do not work on OL8 as it require the removal of python2 support first (some tooling are not available on ol8/python3)
Note:
some tests have been removed (made unresolved) as they require to be fixed to in order
to be run in an automatic manner.

### CODE REVIEW

File: tests/data/commands.xml [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
...mo457lkgbu5s47cia&amp;compartmentId=ocid1.compartment.oc1..aaaaaaaawzn525kzymvcso4rxdfkdgkoscbuqam42dwihe3kzm44iv7mjnta HTTP/1.1\r\nconnection: keep-alive\r\naccept-encoding: gzip, deflate\r\naccept: application/json, application/x-json-stream\r\nuser-agent: Oracle-PythonSDK/2.1.2 (python 2.7.5; x86_64-Linux)\r\nopc-request-id: 5FAA3E9581F9437FA81AE5A0150F77F1\r\nopc-client-info: Oracle-PythonSDK/2.1.2\r\ncontent-type: application/json\r\ndate: Thu, 24 Jan 2019 17:41:34 GMT\r\nhost: iaas.us-ashburn-1.oraclecloud.com\r\nauthorization: Signature algorithm="rsa-sha256",headers="date (request-target) host",keyId="ocid1.tenancy.oc1..aaaaaaaaojlj67jpdjvx7wszzrnwnooss3xxt66oldeyt2x35a6gthqdfr4q/ocid1.user.oc1..aaaaaaaahraqlb5i5c2wdmyjwpqoi5wujry25zjyjio2rlmne6hdvkwwfsba/72:58:bc:dc:8c:62:df:bf:1c:ee:49:f9:88:98:69:e7",signature="EzUXpK+iWu7Zee6dMFUOXRgrc6iDnO89H78myeZY4WQwL+gB8EjogD4FYVFsqvX/CCGeMJVxEUyAE7NtM9gnittfiJw2LA65qhWS94CJ9cLSyp1JaKcGyJ1iGBbJF9ZhsFYuEKjbTn5gfJvoAQaVJfaljuFk0yufq/Y0McONYER/xGwCQwhM0ZSAyRRBk2Cbx8JwX/PeM++KumJIRLagTA1LCHPXM9VXbwOP+Obp04+NG4pKS35WHAqAkXTjjpHYxMeFbd9Tft4CHToEZJRzxMUf0WHiW5WH3+bpwZbxWavn8PNckwCqrIPOo275M2994SRJ7eeWfqqSGBOPSYnJCQ==",version="1"\r\n\r\n'
```
Comment (wbora) [inline tests/data/commands.xml]: I am pretty sure we do not want to commit this to GitHub

File: tests/templates/instance.tfvars.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "tenancy_ocid": "ocid1.tenancy.oc1..aaaaaaaaojlj67jpdjvx7wszzrnwnooss3xxt66oldeyt2x35a6gthqdfr4q",
```
Comment (wbora) [inline tests/templates/instance.tfvars.json]: You really should not be committing this to GitHub

File: tools/provisionning/test_instance/userdata/oci_resolver_config [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown
+# at http://oss.oracle.com/licenses/upl.
+
+search appad2iad.osdiad.oraclevcn.com osdiad.oraclevcn.com us.oracle.com
+nameserver      10.209.76.198
```
Comment (wbora) [inline tools/provisionning/test_instance/userdata/oci_resolver_config]: This is just for internal tenancies, not something to push into GitHub

File: tools/provisionning/test_instance/userdata/proxied_cmd [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
```
Comment (wbora) [inline tools/provisionning/test_instance/userdata/proxied_cmd]: Same comment as above...

File: tests/templates/instance.tfvars.json (hunk 2 above)
Comment (ejannett) [reply tests/templates/instance.tfvars.json]: applied

File: tests/data/commands.xml (hunk 1 above)
Comment (ejannett) [reply tests/data/commands.xml]: deleted

**Pull request merged**